RNA_folding.py

```diff
@@ -0,0 +1,355 @@
+    if not feasible_samples:
+        raise Exception("All solutions infeasible. You may need to try again.")
+        sys.exit()
+
+    # Extract best feasible sample
+    solution = feasible_samples.first
+
+    print('Best Energy:', solution.energy)
+
+    # extract stems with a positive indicator variable.
+    bonded_stems = [stem for stem, val in solution.sample.items() if val == 1 and type(stem) == tuple]
+
+    print('\n# stems in best solution:', len(bonded_stems))
+    print('Stems in best solution:', *bonded_stems)
+
+    if verbose:
+        print('\n# variables (stems):', len(solution[0].keys()))
+
+        # find psuedoknots using product instead of combinations allows for short asymmetric checks.
+        pseudoknots = [(stem1, stem2) for [stem1, stem2] in product(bonded_stems, bonded_stems)
+                       if stem1[1] < stem2[0] and stem2[1] < stem1[2] and stem1[3] < stem2[2]]
+
+        print('\n# pseudoknots in best solution:', len(pseudoknots))
+        if pseudoknots:
+            print('Pseudoknots:', *pseudoknots)
+
+    return bonded_stems
+
+
+# Create command line functionality.
+DEFAULT_PATH = join(dirname(__file__), 'RNA_text_files', 'TMGMV_UPD-PK1.txt')
+
+
+@click.command(help='Solve an instance of the RNA folding problem using '
+                    'LeapHybridCQMSampler.')
+@click.option('--path', type=click.Path(), default=DEFAULT_PATH,
+              help=f'Path to problem file.  Default is {DEFAULT_PATH!r}')
+@click.option('--verbose', is_flag=True, default=True)
+@click.option('--min-stem', type=click.INT, default=3,
+              help='Minimum length for a stem to be considered.')
+@click.option('--min-loop', type=click.INT, default=2,
+              help='Minimum number of nucleotides separating two sides of a stem.')
+@click.option('-c', type=click.FLOAT, default=0.3,
+              help='Multiplier for the coefficient of the quadratic terms for pseudoknots.')
+def main(path, verbose, min_stem, min_loop, c):
+    """ Find optimal stem configuration of an RNA sequence.
+
+    Reads file, creates constrained quadratic model, solves model, and creates a plot of the result.
+    Default parameters are set by click module inputs.
+
+    Args:
+        path: Path to problem file with RNA sequence.
+        verbose: Boolean to determine amount of information printed.
+        min_stem: Integer smallest number of consecutive bonds to be considered a stem.
+        min_loop: Integer minimum number of nucleotides separating two sides of a stem.
+        c: Float multiplier for the coefficient of the quadratic terms for pseudoknots.
+
+    Returns:
+    """
+    # Sanitize inputs
+    assert (min_stem >= 1), 'min_stem must be greater or equal to 1.'
+    assert (min_loop >= 0), 'min_loop must be non-negative.'
+    assert (c >= 0), 'c must be non-negative'
+
+    if verbose:
+        print('\nPreprocessing data from:', path)
+
+    stem_dict = make_stem_dict(text_to_matrix(path), min_stem, min_loop)
```

Comment:
Do you need to pass the value of `min_loop` into `text_to_matrix`?
```suggestion
    stem_dict = make_stem_dict(text_to_matrix(path, min_loop), min_stem, min_loop)
```